### ISSUE TITLE
Sprint 1.29.0 2/2 - Oct 4 to Oct 15

### ISSUE THREAD

**Issue by marcushyett-ph** [PostHog/posthog#6120]
# Global Sprint Planning

Agenda:
* Discussion on ownership @timgl 
* Hangovers from last sprint

## Retro: Status of Outcomes from Previous Sprint

1. **Paths User Experience (Owner: @EDsCODE )** - Design and build user experience for new paths based on customer feedback.
    - Status: @EDsCODE Merged new path card items in Sankey Graph, Basics of Query Builder Implemented - by end of this week - funnel connection, extra path card node - UI tweaks. **1 week worth to go.**
2. **Performance (Owner: @fuziontech )** - Splitting ClickHouse out of helm chart 
    - Status: @fuziontech - De-prioritized because of issues around ingestion and outage (Important but not as urgent as testing the ingestion pipeline) - Materialized cols has bought us a lot of time in terms of performance
3. **Data Integrity** 
    - Ingestion (Owner: @yakkomajuri) - Get to Zero newly created duplicate distinct IDs
        - Status: @yakkomajuri - We're very close to zero - we have a few very unique conditions that still create this - all the teams we're tracking have now got zero
    - Analytics (Owner: @hazzadous) Identify any issues with querying and data integrity based on queries shared
        - Status: @hazzadous One issue which is resolved - looking for help from @paolodamico to validate impact
4. **Session Recording (Owner: @paolodamico)** - Resolve major missing recording and playback issues
     - Status: @paolodamico we had leftover priorities, so did not make significant progress - new session rec page @mariusandra to own
5. **Quantitative Analysis MVP (Owner: @clarkus)** - Design and validate prototype for Quantitative Analysis
    - Status: @clarkus Over-stretched this week - paths was a lot of work, should have a visual prototype to show users by Monday

## Retro: What can we do better next time?
Focussed on our main goal "Paths User Experience" - what can we do better next time? (Add your comments below and we'll discuss during the sprint planning session)

1. Our imagination of the possible edge cases which could occur (might not be possible - focus on quicker feedback cycles between implementation and design)
2. Ship faster and keep behind a flag and iterate on it
3. Making sure we have regular checks to make decisions and centralize them - (Sit down and call the shots - don't wait for consensus in the thread) - Don't wait for consensus - owner call the shots
4. Split out the full UI from functionality, get functionality out as quick as possible to get feedback
5. Keep doing the decision log

## Plan: Proposed Goals for Next Sprint 

Each goal should have a single owner. Owner can only be an engineer.

1. **Quantitative Analysis (Owner: @neilkakkar)** - Basic MVP behind a feature flag by the end of sprint - infront of some specific users (users from paths engagement) - design works in parallel 
    - Why? This is key for people to understand the attributes of successful users and events to diagnose causes
    - **🎯Goal**: _Be able to understand the top 3 properties or events that signal failure in the activation flow of PostHog through a production-ready MVP._
2. **Paths User Experience (Owner: @EDsCODE )** - Get people to use it and more feedback and iterate on it
    - **🎯Goal**: _Get 5 users to use new features for more than 1 day._
2. **Ingestion Data Integrity (Owner: @yakkomajuri)** - Ensuring we can easily track and recover dropped events
    - Why? This remains a critical priority for our customers experiences
3. **Groups** - Interview customers, verifying our assumptions/showing proof of value + understand feature flag + groups story better. Group Analytics Questions #6189 
    - Depending on the results:
         - a. If positive feedback, start implementing, build an MVP with 1-2 client libraries accessible for our team behind a feature flag during the sprint
         - b. If not, back to the drawing board.
4. **Session Recording? (Owner: @mariusandra to assign to another owner e.g. @rcmarron )** - Ship a complete "session recordings" page (and make it easy to find sessions) and an updated playback experience (close open issues - solve the epic)
    - Why? The session recording experience is unreliable and underused - and key to diagnose causes
    - **🎯Goal**: _Two main ingestion errors completely gone ([1](https://sentry.io/organizations/posthog/issues/1726055541/?project=1899813&query=is%3Aunresolved) & [2](https://sentry.io/organizations/posthog/issues/2352164178/?project=1899813&query=is%3Aunresolved)). Note: Consider sentry2 too._
    - **🎯Goal**: _80% of the time I should be able to find the part of the session I care about within 5 seconds. Measured via internal tests._
5. **Refactor and test insights logic? (Owner: @mariusandra)** - Our additive and testless development style has resulted in various insights and dashboards behaving erratically (borked reloads, white screens, etc). It's time to clean house.
    - Why? https://github.com/PostHog/posthog/issues/6118

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
Team ___

## Retro

<!-- Talk about what went well, what didn't go well and any actions to improve next time -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## Planning

<!-- Each item should have a single owner. Owner can only be an engineer. -->

### High priority

-

### Low priority / side quests

-

```

**Comment by paolodamico** [comment]
* **Re recordings.**
      * @mariusandra is now officially the owner for the sessions & recordings epic. Mainly, @rcmarron is working on sessions and @alexkim on recordings.
      * Very limited progress on playback experience improvements. @alexkim mostly worked on leftover work from previous sprint (funnels, saved insights, renaming series, …).
      * We should reprioritize the whole thing for next sprint. This will include design and scoping too.
  * **What can we do better?**
      * Be very deliberate about leftover work and where it fits on priorities, otherwise we won’t be working on the most important things.
      * Test faster with ugly interfaces (though we did gather a lot of user feedback with Figma prototypes, very useful part of the iteration process).
      * Related to the above, if we separate the part of building the high quality UI into a separate PR, we’ll be able to merge the big-functionality-PRs faster, reducing the amount of rework and re-reviews.

**Comment by EDsCODE** [comment]
To be updated..

### Retro + status of hang over items (anything not on this list from last sprint plannig was completed)

Eric: Query builder ui MVP implementation done
Li: Path viz cards MVP done
Neil: Continued path API additions and integrating API with UI
Karl: Benchmarking system for queries in progress (first week was off)
Harry: Squashed several query bugs that made data look wrong
Chris: Paths design work complete as of now. Started looking into quant analysis

What went well: paths work was ironed out and implemented/shipped quickly after decisions were made
What could be better: Set result goals of action items and make sure they're clear and achievable

## Hang over items from previous sprint

1. Prioritized: Enter release phase with paths and iterate with internal user + external users
2. Deprioritized: Day to day work with performance issues will be put in the background. The benchmarking suite will be used but any major changes to increase performance will be tabled for the next sprint atleast

## Planning

High Priority
1. Paths release — Adopt paths as a core insight we use to analyze our users. Get X external users to be power users (Owner: Eric) (Participants: @liyiy 
2. Quantiative Analysis — build MVP (Owner: @neilkakkar) (Participants: @hazzadous, @clarkus )
3. Group Analytics — Draw up exact specs and considerations needed by external users (Owner: @macobo)
4. Performance work — Make sure all queries utilize the new property system when necessary. Benchmark queries (Owner: @macobo)

Low Priority:

- Save cohort of funnel person modal (https://github.com/PostHog/posthog/issues/6086)
- Breakdown by data attr (https://github.com/PostHog/posthog/issues/5914)

**Comment by yakkomajuri** [comment]
## Team Platform

### Retro

We mostly fought fires this sprint. We dealt with things like:

- `createPerson` breaking in the plugin server leading to dropped events
- Heroku Postgres being down
- Poor Kafka partitioning leading to slow ingestion

For the main objectives:

- **Splitting ClickHouse out of helm chart:** Not done
- **Get to Zero newly created duplicate distinct IDs**: We're very close. This is hard to measure but me and @marcushyett-ph did some investigation here and we have barely any new duplicates. 
- **Java library:** We have a working alpha version with the key methods, basic retries, and request batching

We also made good progress with metrics for self-hosted instances - particularly on event volumes and data integrity reporting.

Also pushed some changes to help with data integrity, most notably Kafka messages being sent to S3 automatically.

### Next Up

From the main objectives:

Ingestion Reliability - "Ensuring we can easily track and recover dropped events"

**Comment by marcushyett-ph** [comment]
**Discussion: Performance vs Groups**
Should we reprioritize performance to focus on getting groups live

Why not Performance:
- We've made significant headway - its now less likely to be a reason for churn and disappointment than data integrity for our focus customers

Why Groups:
- This has been asked for by a bunch of clients recently
- The work we did during the hackathon will rapidly become stale if we don't get it live

**Comment by mariusandra** [comment]
Team Core Experience

## Retro

- Rick: not well environment stuff (CH, -js), well: basic session recordings page up (flagg'd)
- Paolo: reprioritise ingestion/onboarding, instead more work on product, finished login, signup 👍 , converations with users, good progress
- Paul: everyone's helping, code in production, didn't break things (yet), lost time with env, good: onboarding week makes it easy to ask questions
- Michael: slight misunderstanding re: scope/architecture of project based permissioning, could/should have done a sync meeting to clear things up before starting coding
- Chris: did some work with onboarding/ingestion (went well), stretch goal for plugins in ingestion = cool, didn't go well: env issues
- Alex: a lot of quality control, renaming filters, win = learning to write kea tests, reviewed many PRs, to improve: sessions recording ownership is confusing

## Hang over items from previous sprint

- Saved insights
- Sessions list
- Session recording playback
- https://github.com/PostHog/posthog/issues/6086

## Planning

### High priority

- Rick: session recordings list/filters/etc https://github.com/PostHog/posthog/pull/6162
- Alex: session recordings playback https://github.com/PostHog/posthog/issues/6077
- Michael: be able to select the project in all API endpoints https://github.com/PostHog/posthog/issues/1343
- Paul: quality improvements for saved insights and other in flight features 
- Paolo: continue preflight/ingestion/onboarding https://github.com/PostHog/posthog/issues/6022
- Chris: support recordings/filters/playback, otherwise analytics work

### Low priority / side quests

- Michael: project ID in the URL for the frontend 
- Rick: toolbar feature flag access rights (snippet vs not) https://github.com/PostHog/posthog/issues/6139
- Paul: continue ingestion bugs?

**Comment by macobo** [comment]
> Discussion: Performance vs Groups

Had a discussion regarding this w/ Marcus.

We're not dropping performance, but not putting 100% focus on it either. At minimum, work there will be finishing the benchmarking suite, do the known optimizations that are backlogged.

Regarding groups: We got a very hacky MVP working in Porto and it makes sense to continue with this before our knowledge rots. Before rushing this in front of our users though, we'll take a step back and validate if building this with the underlying assumptions we did makes sense. 

The plan is:
1. Interview customers, verifying our assumptions/showing proof of value + understand feature flag + groups story better. https://github.com/PostHog/posthog/issues/6189 @marcushyett-ph and @macobo 
2. Depending on the results:
   a. If positive feedback, start implementing, build an MVP with 1-2 client libraries accessible for our team behind a feature flag during the sprint
   b. If not, back to the drawing board.
